src/main/scala/com/typesafe/sbt/packager/jdkpackager/JDKPackagerAntHelper.scala

```diff
@@ -0,0 +1,246 @@
+package com.typesafe.sbt.packager.jdkpackager
+
+import com.typesafe.sbt.packager.jdkpackager.JDKPackagerKeys.{ FileAssociation, JDKPackagerToolkit }
+import org.apache.tools.ant.{ BuildEvent, BuildListener, ProjectHelper }
+import sbt.Keys._
+import sbt._
+
+import scala.util.Try
+
+/**
+ * Helpers for working with Ant build definitions
+ *
+ * @author <a href="mailto:fitch@datamininglab.com">Simeon H.K. Fitch</a>
+ * @since 5/7/15
+ */
+object JDKPackagerAntHelper {
+
+  /** Attempts to compute the path to the `javapackager` tool. */
+  private[jdkpackager] def locateAntTasks(javaHome: Option[File]): Option[File] = {
+    val jarname = "ant-javafx.jar"
+
+    // This approach to getting JDK bits is borrowed from: http://stackoverflow.com/a/25163628/296509
+    // Starting with an ordered list of possible java directory sources, create derivative and
+    // then test for the tool. It's nasty looking because there's no canonical way of finding the
+    // JDK from the JRE, and JDK_HOME isn't always defined.
+    Seq(
+      // Build-defined
+      javaHome,
+      // Environment override
+      sys.env.get("JDK_HOME").map(file),
+      sys.env.get("JAVA_HOME").map(file),
+      // MacOS X
+      Try("/usr/libexec/java_home".!!.trim).toOption.map(file),
+      // From system properties
+      sys.props.get("java.home").map(file)
+    )
+      // Unlift Option-s
+      .flatten
+      // For each base directory, add the parent variant to cover nested JREs on Unix.
+      .flatMap {
+        f ⇒ Seq(f, f.getAbsoluteFile)
+      }
+      // On Windows we're often running in the JRE and not the JDK. If JDK is installed,
+      // it's likely to be in a parallel directory, with the "jre" prefix changed to "jdk"
+      .flatMap { f ⇒
+        if (f.getName.startsWith("jre")) {
+          Seq(f, f.getParentFile / ("jdk" + f.getName.drop(3)))
+        } else {
+          Seq(f)
+        }
+      }
+      // Now search for the tool
+      .map { n ⇒
+        n / "lib" / jarname
+      }
+      .find(_.exists)
+  }
+
+  /**
+   * Create Ant project DOM for building packages, using ant-javafx.jar tasks.
+   *
+   * see: https://docs.oracle.com/javase/8/docs/technotes/guides/deploy/javafx_ant_task_reference.html
+   */
+  private[jdkpackager] def makeAntBuild(
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The number of parameters should not exceed 8](https://www.codacy.com/app/nepomukseiler/sbt-native-packager/file/637524688/issues/source?bid=2040925&fileBranchId=2311583#l64)